src/addressing_modes.rs

```diff
@@ -79,14 +80,15 @@ pub(crate) const SLOAD_COST: u32 = 2008;
 pub(crate) const INVALID_INSTRUCTION_COST: u32 = 4294967295;
 
 impl Arguments {
-    pub const fn new(predicate: Predicate, gas_cost: u32) -> Self {
+    pub const fn new(predicate: Predicate, gas_cost: u32, requires_kernel_mode: bool) -> Self {
```

Comment:
Please check kernel mode in the relevant instructions instead of storing it in arguments.